### DISCUSSION TITLE
device_class: vibration

### DISCUSSION THREAD

**Discussion by grinco** [jseidl/magic-areas#192] [Ideas / Feature Requests]
I'm using a vibration sensor on my chair to detect presence. Currently, I need to have an occupancy template sensor that changes states based on the vibration sensor for the presence to be detected. Can you maybe add native vibration sensor support? Would that make sense for other people?

**Comment by Patrick1610** [comment]
What a coincidence, I have the exact same use for vibration sensors and will be expanding that throughout my house! I would love this support! 🙏🏼

**Comment by grinco** [comment]
If it helps - here's how I'm handling it now:

```
platform: template
sensors:
emilys_chair_presence:
device_class: occupancy
value_template: >-
{{is_state('binary_sensor.emily_s_chair_vibration', 'on') }}
jankas_chair_presence:
device_class: occupancy
value_template: >-
{{is_state('binary_sensor.janka_s_chair_vibration', 'on') }}
```

**Comment by jseidl** [comment]
All `binary_sensor` device classes are being exposed on the 3.0.0 release. If you folks are brave you can try it out, currently on beta. https://github.com/jseidl/hass-magic_areas/tree/release/3.0.0-beta

**Comment by Patrick1610** [comment]
That's great news! I really have a lot of automations that rely on areas, either directly or indirectly, so I'm not brave enough to use the beta! Do you have any idea when it wil go from beta to release?

**Comment by jseidl** [comment]
I've been testing for a while and seems to be stable enough. I've just added one last feature to the beta and will be testing for a few more weeks before I can release it. Unfortunately it is not backwards compatible which means users will have to reconfigure all areas but it's a huge improvement from the 2.x release in terms of features and performance+reliability.

**Comment by Patrick1610** [comment]
That's great! I have been wanting to change some things to my areas so starting over is no problem! As soon as the entities are called the same automations will continue to work, right? 
Will I be able to setup 3.0 before losing all the areas from 2.x or is it an upgrade for 2.x? 
Will it be possible to properly use subareas in areas? If I recall correctly that didn't work in the previous version (at the time I implemented it). For example: make the area 'Toilet' an entity in area 'Hallway', as long as the toilet is active the hallway is too.

**Comment by jseidl** [comment]
> Will I be able to setup 3.0 before losing all the areas from 2.x or is it an upgrade for 2.x?
So far I haven't implemented logic for dealing with 2.x configuration in 3.x, but will give a whirl 

> Will it be possible to properly use subareas in areas?
It should be. I haven't tested but you should be able to add one Magic Area's entity to another area.

**Comment by Patrick1610** [comment]
> > Will I be able to setup 3.0 before losing all the areas from 2.x or is it an upgrade for 2.x?
> 
> So far I haven't implemented logic for dealing with 2.x configuration in 3.x, but will give a whirl 

How is it coming along? If it's possible to test 3.0 while 2.x is still running and you need some testers I can try to get it going in my home assistant 🤗 or are you so close to release that it's not necessary anymore?

> > Will it be possible to properly use subareas in areas?
> 
> It should be. I haven't tested but you should be able to add one Magic Area's entity to another area. 

Have you gotten around to testing this? I can play around with that myself when 3.0 is installed as well, so I'll let you know for sure when I have tested that 🙋🏼‍♂️

**Comment by jseidl** [comment]
Hi! 3.0 is just out, this is currently present on the latest release!

**Reply by grinco** [reply]
Do I need to follow any process to upgrade! All my entities became unavailable after the update.

**Reply by grinco** [reply]
Nevermind. Deleting all zones and restarting HA did the job. Too bad. I had custom configuration on almost all of them.

**Reply by jseidl** [reply]
Yes unfortunately 3.0 had some breaking changes (stated in the release notes) so all the areas needs to be recreated. Sorry about that, there was no other way as `unique_id`s had to be changed along with the configuration storage.

**Reply by grinco** [reply]
That's quite all right! Good job on the integration. I really love it. Lights controls work well. I haven't tested some of the more advanced functionality - like the "follow me" media player, or the new vibration sensor. I'll be happy to report any hugs if any. Thanks for investing your time into it.
